### PR TITLE
fix(UI): Error when creating/editing duplicate project/component/release

### PR DESCRIPTION
**Pull request by afsahsyeda**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

The edit page is retained after 'DUPLICATE' error message is thrown.

Issue: #957
Error on creating/editing duplicate project(existing name and version).

### How To Test?

Create:
- Add a new `project`/`component`/`release` with the same name and version as an existing `project`/`component`/`release`.
- Click on Create `project`/`component`/`release` button and check if all the `project`/`component`/`release` related information is retained (**no data loss**).
- In case of project, please add `linked release` and `projects`, and check if all the `linked releases` and `sub-projects` information are retained in case of duplicate error  (**no `linked releases` and `sub-projects` loss**).

Edit:
- Edit an existing `project`/`component`/`release` and change the name and version to a different (an existing `project`/`component`/`release`).
- Click on Update `project`/`component`/`release` button and check if all the `project`/`component`/`release` related information is retained (**no data loss**).
- Please add new attachments / remove existing attachments and check if all the attachment related information is retained.

An error stating "**`project`/`component`/`release` with same name and version already exists**" will pop up but the edit page should retain with user entered data.
image

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: afsahsyeda <afsah.syeda@siemens-healhtineers.com>
Signed-off-by: akapti <abdul.kapti@siemens-healhtineers.com>

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Testing was successful.

**Review by akapti — CHANGES_REQUESTED** [review]
One issue I noticed while testing is that linked sub-projects and attachments are not retained. 
image


All the data entered by user is retained except linked sub-projects (creating a duplicate project or editing an existing project) and attachments (only when editing existing project).

**Comment by akapti** [comment]
One more thing... The exact same issue is present for Component as well.
image


It would be really great if the component issue can also be fixed or create a new issue for that and provide a fix in another PR.

**Comment by afsahsyeda** [comment]
Hi @akapti ,

Please check for similar changes implemented in the Component portlet.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 1]
```diff
@@ -2242,20 +2242,27 @@ private void prepareProjectEdit(RenderRequest request) {
...
-
             try {
-                ProjectService.Iface client = thriftClients.makeProjectClient();
-                project = client.getProjectByIdForEdit(id, user);
+                if (project == null) {
+                    project = client.getProjectByIdForEdit(id, user);
+                } else {
+                    isUpdateOrCreateProjectFailed = true;
+                }
+                project = project == null ? client.getProjectByIdForEdit(id, user) : project;
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:2265]: Is there any significant of this line?

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Requesting for some minor validations

**Comment by afsahsyeda** [comment]
Hi @smrutis1 ,
 I have reused the OBJECT_MAPPER instance and removed ternary expression in ProjectPortlet as I did not see any change after removing the line.

**Review by smrutis1 — APPROVED** [review]
Code looks ok

**Pull request merged**